xbmc/addons/kodi-addon-dev-kit/CMakeLists.txt

```diff
@@ -2,6 +2,17 @@ project(kodi-addon-dev-kit)
 
 cmake_minimum_required(VERSION 2.8.7)
 
+# add a target to generate API documentation with Doxygen
+find_package(Doxygen)
+
+if(DOXYGEN_FOUND)
+  add_custom_target(doc
+    ${DOXYGEN_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/doxygen/Doxyfile
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+    COMMENT "Generating API documentation with Doxygen" VERBATIM
+  )
+endif(DOXYGEN_FOUND)
+
```

Comment:
It is used by the code generator to supply documentation in the generated python modules.